### PR TITLE
PSFR-855 Update dependencies and image

### PR DESCRIPTION
**Pull request by mkerry-moj**
* Java 19 -> 21 - his enables upgrade to hmpps-sqs-spring-boot-starter and the older version that supported java 19 came with security issues.
* Upgrade dependencies

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt [hunk 1]
```diff
@@ -34,11 +36,13 @@ class EndpointSecurityCheck {
...
   }
 
+  private val excluded = setOf(HmppsQueueResource::class.java, HmppsReactiveQueueResource::class.java)
+
   private fun getAllUnprotectedControllers() = ClassPathScanningCandidateComponentProvider(false)
     .also { it.addIncludeFilter(AnnotationTypeFilter(RestController::class.java)) }
     .findCandidateComponents("uk.gov.justice")
     .map { Class.forName(it.beanClassName) }
-    .filter { !it.isProtectedByAnnotation() }
+    .filter { !it.isProtectedByAnnotation() && !excluded.contains(it) }
```
Comment (JamesBoobier) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt:45]: Not sure these should be excluded as they should be behind the RESETTLEMENT_PASSPORT_EDIT role

Comment (mkerry-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt:45]: They are endpoints for managing dead letter queues exposed automatically by the hmpps-sqs-spring-boot-starter. Did they have it before and have lost it somehow or are they new endpoints (which was my assumption)

Comment (mkerry-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt:45]: Either way - I can't inject the annotation on them, I'd have to add the routes specifically to the `ResourceServerConfiguration` file so the test would have to be separate

Comment (JamesBoobier) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt:45]: These are existing endpoints that were set up so we can intergation with SQS in the future. You can set the role in the application.yml: https://github.com/ministryofjustice/hmpps-resettlement-passport-api/blob/main/src/main/resources/application.yml#L118 so not sure why that has stopped working...

Comment (mkerry-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/resource/EndpointSecurityCheck.kt]: 👍

**Review by JamesBoobier — APPROVED** [review]

**Pull request merged**